### PR TITLE
Maintains any error message if LoginCallback has an error

### PR DESCRIPTION
**Pull request by swiftone**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our [guidelines](/okta/okta-oidc-js/blob/master/CONTRIBUTING.md#commit)
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Adding Tests
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation changes
- [ ] Other... Please describe:

## What is the current behavior?

Currently a pending async initial authState determination can replace any error message shown in LoginCallback from trying to handle authentication of the tokens in the URL.

Issue Number: #719

## What is the new behavior?

Now LoginCallback will wait until the initial authState determination is complete before triggering the handleAuthentication or showing any error.

## Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

File: packages/okta-react/src/LoginCallback.js [hunk 1]
```diff
@@ -12,16 +12,23 @@
...
-  }, [authService]);
+    if( authStateReady ) { // wait until initial check is done so it doesn't wipe any error
+      authService.handleAuthentication();
+    }
+  }, [authService, authStateReady]);
 
-  if(authState.error) { 
-    return <p>{authState.error.toString()}</p>;
+  if(authStateReady && authState.error) { 
+    // return errorReporter(authState.error);
```
Comment (aarongranick-okta) [inline packages/okta-react/src/LoginCallback.js:30]: intentional comment?

File: packages/okta-react/src/OktaError.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import React from 'react';
+
+const OktaError = ({ error }) => { 
+  if(error.name && error.message) { 
+    return <p>{error.name}: {error.message}</p>;
```
Comment (aarongranick-okta) [inline packages/okta-react/src/OktaError.js:17]: why don't we want to use error.toString()? Doesn't it produce the same result?

File: packages/okta-react/src/LoginCallback.js (hunk 1 above)
Comment (swiftone) [reply packages/okta-react/src/LoginCallback.js:30]: :rage: No, thanks.

File: packages/okta-react/src/OktaError.js (hunk 2 above)
Comment (swiftone) [reply packages/okta-react/src/OktaError.js:17]: Not in one case, at least. (I was seeing 'OktaApiError' but no text), plus this pseudo-documents how someone would write their own override.

**Pull request merged**